crates/storage/provider/src/test_utils/blocks.rs

```diff
@@ -0,0 +1,130 @@
+
+use crate::{
+    execution_result::{
+        AccountChangeSet, AccountInfoChangeSet, ExecutionResult, TransactionChangeSet,
+    },
+    Transaction,
+};
+use reth_db::{database::Database, models::StoredBlockBody, tables};
+use reth_primitives::{
+    hex_literal::hex, proofs::EMPTY_ROOT, Account, Header, Receipt, SealedBlock,
+    SealedBlockWithSenders, Withdrawal, H160, H256, U256,
+};
+use reth_rlp::Decodable;
+use std::collections::BTreeMap;
+
+/// Assert genesis block
+pub fn assert_genesis_block<DB: Database>(tx: &Transaction<'_, DB>, g: SealedBlock) {
+    let n = g.number;
+    let h = H256::zero();
+    // check if all tables are empty
+    assert_eq!(tx.table::<tables::Headers>().unwrap(), vec![(g.number, g.header.clone().unseal())]);
+
+    assert_eq!(tx.table::<tables::HeaderNumbers>().unwrap(), vec![(h, n)]);
+    assert_eq!(tx.table::<tables::CanonicalHeaders>().unwrap(), vec![(n, h)]);
+    assert_eq!(tx.table::<tables::HeaderTD>().unwrap(), vec![(n, g.difficulty.into())]);
+    assert_eq!(tx.table::<tables::BlockBodies>().unwrap(), vec![(0, StoredBlockBody::default())]);
+    assert_eq!(tx.table::<tables::BlockOmmers>().unwrap(), vec![]);
+    assert_eq!(tx.table::<tables::BlockWithdrawals>().unwrap(), vec![]);
+    assert_eq!(tx.table::<tables::Transactions>().unwrap(), vec![]);
+    assert_eq!(tx.table::<tables::TxHashNumber>().unwrap(), vec![]);
+    assert_eq!(tx.table::<tables::Receipts>().unwrap(), vec![]);
+    assert_eq!(tx.table::<tables::PlainAccountState>().unwrap(), vec![]);
+    assert_eq!(tx.table::<tables::PlainStorageState>().unwrap(), vec![]);
+    assert_eq!(tx.table::<tables::AccountHistory>().unwrap(), vec![]);
+    assert_eq!(tx.table::<tables::StorageHistory>().unwrap(), vec![]);
+    // Bytecodes are not reverted assert_eq!(tx.table::<tables::Bytecodes>().unwrap(), vec![]);
+    assert_eq!(tx.table::<tables::BlockTransitionIndex>().unwrap(), vec![(n, 0)]);
+    assert_eq!(tx.table::<tables::TxTransitionIndex>().unwrap(), vec![]);
+    assert_eq!(tx.table::<tables::AccountChangeSet>().unwrap(), vec![]);
+    assert_eq!(tx.table::<tables::StorageChangeSet>().unwrap(), vec![]);
+    assert_eq!(tx.table::<tables::HashedAccount>().unwrap(), vec![]);
+    assert_eq!(tx.table::<tables::HashedStorage>().unwrap(), vec![]);
+    assert_eq!(tx.table::<tables::AccountsTrie>().unwrap(), vec![(EMPTY_ROOT, vec![0x80])]);
+    assert_eq!(tx.table::<tables::StoragesTrie>().unwrap(), vec![]);
+    assert_eq!(tx.table::<tables::TxSenders>().unwrap(), vec![]);
+    // SyncStage is not updated in tests
+}
+
+/// Genesis block
+pub fn genesis() -> SealedBlock {
+    SealedBlock {
+        header: Header { number: 0, difficulty: U256::from(1), ..Default::default() }
+            .seal(H256::zero()),
+        body: vec![],
+        ommers: vec![],
+        withdrawals: Some(vec![]),
+    }
+}
+
+/// Block one that points to genesis
+pub fn block1() -> (SealedBlockWithSenders, ExecutionResult) {
```

Comment:
Tthis seems to only be used in one file, so I would move it near that file instead. `block1` is not a super useful name for a general purpose test utility

I would also prob clarify what the `ExecutionResult` is here

Reply:
It is used in provider::Transaction and in blockchain tree. Added a new structure that wraps this around.